src/parsing/parser.h

```diff
@@ -11,13 +11,30 @@ namespace v8 {
 namespace internal {
 
 template <>
-struct ParserTypes<Parser> : AbstractParserTypes<Parser> {
-};
+struct ParserTypes<Parser> : AbstractParserTypes<Parser> {};
 
 class V8_EXPORT_PRIVATE Parser
     : public NON_EXPORTED_BASE(AbstractParser<Parser>) {
  public:
   Parser(ParseInfo* info) : AbstractParser<Parser>(info) {}
+
+ private:
+  FunctionLiteral* DoParseFunction(Isolate* isolate, ParseInfo* info,
+                                   int start_position, int end_position,
+                                   int function_literal_id,
+                                   const AstRawString* raw_name) {
+    DCHECK_EQ(impl()->parsing_on_main_thread_, isolate != nullptr);
+
+    return AbstractParser<Parser>::DoParseFunction(
+        isolate, info, start_position, end_position, function_literal_id,
+        raw_name);
+  }
+
+  void PostProcessParseResult(Isolate* isolate, ParseInfo* info,
+                              FunctionLiteral* literal) {
+    DCHECK_EQ(isolate != nullptr, impl()->parsing_on_main_thread_);
+    AbstractParser<Parser>::PostProcessParseResult(isolate, info, literal);
```

Comment:
is this the right way to call the "super" class' method?

Reply:
Yep, I believe so!